### PR TITLE
Kernel/ScatterGatherList: Improve error propagation

### PR DESCRIPTION
**Pull request by Panky-codes**
A small cleanup to remove some of the TODOs in the ScatterGatherList for better error propagation.

### CODE REVIEW

**Review by BenWiederhake — APPROVED** [review]
Nicely factored commits, does what it says on the tin: It improves our error propagation. The call to try_create is still immediately followed by a `release_value_but_fixme_should_propagate_errors`, so there's no immediate gain from this (any error would still cause an assert), but this is definitely progress, so let's merge it. (And then hopefully continue connecting everything up.)

**Comment by Panky-codes** [comment]
Thanks @BenWiederhake! `AHCI` needs some refactoring to remove the `release_value_but_fixme_should_propagate_errors` to properly propagate errors but that should be a part of a separate refactoring effort that I might do soon. In the meantime, this API can be used with proper error propagation by other drivers :) .

File: Kernel/Memory/ScatterGatherList.cpp [hunk 1]
```diff
@@ -8,11 +8,11 @@
 
 namespace Kernel::Memory {
 
-ErrorOr<LockRefPtr<ScatterGatherList>> ScatterGatherList::try_create(AsyncBlockDeviceRequest& request, Span<NonnullRefPtr<PhysicalPage>> allocated_pages, size_t device_block_size)
+ErrorOr<LockRefPtr<ScatterGatherList>> ScatterGatherList::try_create(AsyncBlockDeviceRequest& request, Span<NonnullRefPtr<PhysicalPage>> allocated_pages, size_t device_block_size, StringView region_name)
 {
     auto vm_object = TRY(AnonymousVMObject::try_create_with_physical_pages(allocated_pages));
     auto size = TRY(page_round_up((request.block_count() * device_block_size)));
-    auto region = TRY(MM.allocate_kernel_region_with_vmobject(vm_object, size, "AHCI Scattered DMA"sv, Region::Access::Read | Region::Access::Write, Region::Cacheable::Yes));
+    auto region = TRY(MM.allocate_kernel_region_with_vmobject(vm_object, size, region_name, Region::Access::Read | Region::Access::Write, Region::Cacheable::Yes));
```
Comment (GMTA) [inline Kernel/Memory/ScatterGatherList.cpp:15]: This entire commit should probably go into the last one. You hardcoded the region's name in the last commit, so you're "fixing" it by adding another commit on top. That's a good indicator that it should have been combined in the first place :^)

**Review by GMTA — CHANGES_REQUESTED** [review]
Nice!

File: Kernel/Memory/ScatterGatherList.cpp (hunk 1 above)
Comment (Panky-codes) [reply Kernel/Memory/ScatterGatherList.cpp:15]: Nope, the hard-coded string was already there.

Comment (GMTA) [reply Kernel/Memory/ScatterGatherList.cpp:15]: Oh you're right! Apologies, then it makes sense like this :^)

**Review by GMTA — APPROVED** [review]

**Pull request merged**